kg_microbe/transform_utils/bacdive/bacdive.py

```diff
@@ -1042,18 +1048,89 @@ def run(self, data_file: Union[Optional[Path], Optional[str]] = None, show_statu
                                 ),
                                 None,
                             )
-                            if ncbi_of_interest is not None:
-                                ncbitaxon_id = NCBITAXON_PREFIX + str(ncbi_of_interest)
+                            if selected_ncbi is not None:
+                                ncbitaxon_id = NCBITAXON_PREFIX + str(selected_ncbi[NCBITAXON_ID])
                         else:
                             ncbitaxon_id = NCBITAXON_PREFIX + str(
                                 general_info[NCBITAXON_ID][NCBITAXON_ID]
                             )
-                        species_with_strains = [ncbitaxon_id]
                         ncbi_description = general_info.get(GENERAL_DESCRIPTION, "")
                         ncbi_label = get_label(self.ncbi_impl, ncbitaxon_id)
                         if ncbi_label is None:
                             ncbi_label = ncbi_description
 
+                    # If no NCBITaxon ID from BacDive API, try searching by name
```

Comment:
Minor: not BacDive API –> BacDive JSON (may be constructed using API not sure but we're not consuming API directly)